### PR TITLE
Add upload archives workflow

### PR DESCRIPTION
**Pull request by rqbacktrace**
Enable GitHub actions to upload archives to Maven when a version tag is pushed

### CODE REVIEW

File: .github/workflows/uploadArchives.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ push:
+   tags: '[0-9]+.[0-9]+.[0-9]+.*'
+
+jobs:
+ build:
+   env: 
+     ANDROID_API_LEVEL: 23
+     ANDROID_EMULATOR_LEVEL: 23
+     ANDROID_BUILD_TOOLS_VERSION: 28.0.3
+     ANDROID_ABI: armeabi-v7a
```
Comment (konraddysput) [inline .github/workflows/uploadArchives.yml:13]: not sure if this is important, but why we didn't include the arm64 support here?

File: .github/workflows/uploadArchives.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     with:
+       fetch-depth: 2
+       submodules: recursive
+       token: ${{ secrets.ACCESS_TOKEN }}
+       persist-credentials: false
+
+   - name: Grant execute permission for gradlew
+     run: chmod +x gradlew
+   - name: Upload archives
+     run: ./gradlew uploadArchives
```
Comment (konraddysput) [inline .github/workflows/uploadArchives.yml:39]: not sure if I know where is the uploadArchives job code. Can you give me a reference to the code?

**Review by konraddysput — COMMENTED** [review]
Some questions that confirm my understanding of the problem. Seeking from help from someone.

**Comment by rqbacktrace** [comment]
@BartoszLitwiniuk @vlussenburg @konraddysput I have tested this and it works.

Here is the link to the successful workflow run:
https://github.com/backtrace-labs/backtrace-android/runs/4873714656

Here is a screenshot of the build uploaded to SonaType with timestamp
AutoUploadSuccessful
:

File: .github/workflows/uploadArchives.yml (hunk 1 above)
Comment (rqbacktrace) [reply .github/workflows/uploadArchives.yml]: @konraddysput I removed the `ANDROID_ABI` environment variable, it's not needed here anyway.

File: .github/workflows/uploadArchives.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+name: Upload Archives
+
+on:
+ push:
+   tags: '[0-9]+.[0-9]+.[0-9]+.*'
+ workflow_dispatch:
```
Comment (BartoszLitwiniuk) [inline .github/workflows/uploadArchives.yml:6]: do we need that? because it is empty

**Review by BartoszLitwiniuk — APPROVED** [review]
Good job @rqbacktrace !

**Pull request merged**